internal/installutil/adoption.go

```diff
@@ -0,0 +1,69 @@
+package installutil
+
+import (
+	"context"
+	"errors"
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"time"
+)
+
+func AdoptRunner(ctx context.Context, ui terminal.UI, client pb.WaypointClient, id string, addr string) error {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Waiting for runner to connect to server at %s...", addr)
+	defer func() { s.Abort() }()
+
+	// Waits 5 minutes for the server to detect the new runner before timing out
+	d := time.Now().Add(time.Minute * time.Duration(5))
+	ctx, cancel := context.WithDeadline(ctx, d)
+	defer cancel()
+	ticker := time.NewTicker(5 * time.Second)
+	found := false
+	for !found {
+		select {
+		case <-ticker.C:
+		case <-ctx.Done():
+			ui.Output("Cancelled.",
+				terminal.WithErrorStyle(),
+			)
+			return errors.New("context canceled")
+		}
+		// Use runner list API to check if runner is reporting to server yet
+		// If it's found, adopt it. Otherwise, try until deadline.
+		runners, err := client.ListRunners(ctx, &pb.ListRunnersRequest{})
+		if err != nil {
+			ui.Output("Error getting runners: %s", clierrors.Humanize(err),
+				terminal.WithErrorStyle(),
+			)
+			return err
+		}
+		for _, myRunner := range runners.Runners {
+			if myRunner.Id == id {
+				found = true
```

Comment:
@catsby I went with using `found` here since I'm using `break` on [line 46](https://github.com/hashicorp/waypoint/blob/4ea71702338d96cd41997521ee2a77ba4564b13f/internal/installutil/adoption.go#L46), that breaks us out of this loop, but not the "parent" loop started on [line 25](https://github.com/hashicorp/waypoint/blob/4ea71702338d96cd41997521ee2a77ba4564b13f/internal/installutil/adoption.go#L25).